platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/Contexts.kt

```diff
@@ -332,11 +297,85 @@ class DefaultReadContext(
     }
 
     private
-    fun readHashCode() = if (readBoolean()) {
+    fun ReadContext.readHashCode() = if (readBoolean()) {
         HashCode.fromBytes(readBinary())
     } else {
         null
     }
+}
+
+
+internal
```

Comment:
Feedback for future PRs: this PR is a bit harder to follow than needed due to code moves mixed with actual changes (the class below was not added, for instance).

Reply:
New classes were added before that one with some extracted code and the diff got mangled somehow.

Reply:
I ended up reordering stuff locally and the diff got much saner

Reply:
For sure, but at least for non-trivial change sets, I try to delay form-related changes to a follow-up form-only PR.

Reply:
Ok. It's a valid choice.

I see it like this: new types were added.  It seems you are suggesting they should be added to the bottom of the file for simpler diff first and then later moved (in a separate PR) to their intended location. I think, you, the reader, can deal with the additional diff complexity so I, the writer, don't have to come up with two PRs just to have new classes in their intended location.